spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/RetryGatewayFilterFactory.java

```diff
@@ -161,7 +167,7 @@ public GatewayFilter apply(Repeat<ServerWebExchange> repeat, Retry<ServerWebExch
 		
 		private List<HttpMethod> methods = toList(HttpMethod.GET);
 
-		private List<Class<? extends Throwable>> exceptions = toList(IOException.class);
+		private List<Class<? extends Throwable>> exceptions = toList(IOException.class, TimeoutException.class);
```

Comment:
Is this a reason that this, and all the `toList()` calls isn't just `Arrays.asList()`?

Reply:
I wanted a mutable list

Reply:
Fair enough.  I'd implement a single `toList()` method then for both the single and double since the new one creates an unnecessary object.

```java
private static <T> List<T> toList(T... items) {
	ArrayList<T> list = new ArrayList<>();
    for(T item: items) {
	    list.add(item);
    }
	return list;
}
```

Reply:
Yeah, I already have a similar change already implemented.